modules/context/context.go

```diff
@@ -348,7 +349,9 @@ func (ctx *Context) RespHeader() http.Header {
 // SetServeHeaders sets necessary content serve headers
 func (ctx *Context) SetServeHeaders(filename string) {
 	ctx.Resp.Header().Set("Content-Description", "File Transfer")
-	ctx.Resp.Header().Set("Content-Type", "application/octet-stream")
+	if _, has := ctx.Resp.Header()[textproto.CanonicalMIMEHeaderKey("Content-Type")]; !has {
```

Comment:
how about `len(ctx.Resp.Header().Values("Content-Type")) == 0` ? it looks a little simpler.

* Non-existing key: `ctx.Resp.Header().Values` returns zero-length slice.
* Existing key: `ctx.Resp.Header().Values` returns nonzero-length slice.

Reply:
That's only a workaround until #21833 is merged. I will mark this PR as blocked and change it with the merge then.